src/com/company/Entities/Theatre.java

```diff
@@ -0,0 +1,103 @@
+package com.company.Entities;
+
+import com.company.Events.Event;
+
+import java.util.*;
+
+public class Theatre {
+    //singleton (early initialization)
+    public static final Theatre theatre = new Theatre();
+
+    private String name = "HappyTickets"; // TODO read from file
+    private String address = "Street no. 5, West";  // TODO read from file
+    private Map<String, String> openHours = Map.of("Monday", "10-18", "Tuesday", "11-18");
+    private List<Hall> halls; //TODO take from file
+    private List<Event> futureEvents, pastEvents;
+
+    private Theatre(){
+        halls = new ArrayList<Hall>();
+        futureEvents = new ArrayList<Event>();
+        pastEvents = new ArrayList<Event>();
+    }
+
+    public static Theatre getTheatre() {
+        return theatre;
+    }
+
+    public List<Event> getIncomingEvents() {
+        return futureEvents;
+    }
+
+    public void setIncomingEvents(List<Event> incomingEvents) {
+        this.futureEvents = incomingEvents;
+    }
+
+    public void showTheatreInformation() {
+        System.out.println("Welcome to " + name + "! We are happy to see you here :)");
+        System.out.println("Address: " + address);
+        System.out.println("Opening hours:");
+        for (Map.Entry mapElement : openHours.entrySet()) {
+            System.out.print(mapElement.getKey() + ": " +  mapElement.getValue() + "  ");
+        }
+        System.out.println();
+    }
+
+    public boolean addEvent(Event event){
+        //TODO READ INFO AND ADD TO FILE HERE
+        if(event != null) {
+            this.futureEvents.add(event);
+            return true; //done
+        }
+        return false; //something went wrong

```

Comment:
pe viitor poti inlocui cu aruncarea unei exceptii custom in cazul in care eventul este null :)